src/Parser/Template.cs

```diff
@@ -54,6 +57,95 @@ public Template(string template, bool isPositional, List<Literal> literals, List
       Holes = holes.ToArray();
     }
 
+    public string Render(object[] parameters)
+    {
+        var sb = new StringBuilder();
```

Comment:
The default `16` is clearly too small (I didn't know it was _that_ small!).
After that it's really wild guesses because any hole can be tiny or huge.

Could we use `Value.Length + constant * Holes.Length` for some arbitrary constant?

The thing with `Value.Length * 2` is that we might be overshooting by a lot for long texts with few placeholders (is that bad?), e.g. `Process {name} completed successfully after {time}ms.` is 53 char, so it would allocate 212 bytes, when it probably only needs half as much.
Of course with very long holes all bets are off, like: `Result = {@fat}`, which is tiny and won't allocate much but can end up requiring _a lot_ of memory.